src/org/mirah/jvm/mirrors/mirror_type_system.mirah

```diff
@@ -744,6 +751,46 @@ class MirrorTypeSystem implements TypeSystem
     c = a.widen(b)
     puts c
   end
+
+  def register_extensions:void
+    boot_class_loader = MirrorTypeSystem.class.getClassLoader
+    register_extensions boot_class_loader
+    
+    compile_class_loader = ClassLoader(@context[ClassLoader])
+    
+    if compile_class_loader != boot_class_loader
+      register_extensions compile_class_loader
+    end
+  end
+  # use java service SPI to load all extensions registrations from context classloader
+  def register_extensions(class_loader:ClassLoader):void    
+    return unless class_loader
+    services = ServiceLoader.load(ExtensionsProvider.class, class_loader)
+    type_system = self
+    iter = Iterable(services).iterator
+    while iter.hasNext
+      provider = ExtensionsProvider(iter.next)
+      provider.register(type_system)
+    end
+  rescue Error => ex
+    puts "Unable to load regtistration service: #{ex}"
```

Comment:
We really don't want to rescue Error here. How about ` ServiceConfigurationError `? It's the thing that http://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html#iterator() raises.

Also, I think we should re-raise in the error case, because it's a compiler misconfiguration.